### ISSUE TITLE
Broken stack if namespace deletion fails

### ISSUE THREAD

**Issue by manast** [pulumi/pulumi#5344]
This happened for a kubernetes namespace but I report it here because I do not think this problem is specific for kubernetes.
So in this case the namespace did not get deleted for an unknown reason when doing pulumi destroy:
```
 -   ├─ kubernetes:core:Namespace                                    contour                                                            **deleting failed**     2 errors
```

```
 kubernetes:core:Namespace (contour):
    error: Timeout occurred polling for 'contour'
    error: post-step event returned an error: failed to save snapshot: [403] The provided update token has expired.
```

Now when retrying "pulumi destroy" we get this error:
```
error: the current deployment has 1 resource(s) with pending operations:
  * urn:pulumi:dev::taskforce-infra::taskforce:project$taskforce:region$taskforce:contour$kubernetes:core/v1:Namespace::contour, interrupted while deleting

These resources are in an unknown state because the Pulumi CLI was interrupted while
waiting for changes to these resources to complete. You should confirm whether or not the
operations listed completed successfully by checking the state of the appropriate provider.
For example, if you are using AWS, you can confirm using the AWS Console.

Once you have confirmed the status of the interrupted operations, you can repair your stack
using 'pulumi stack export' to export your stack to a file. For each operation that succeeded,
remove that operation from the "pending_operations" section of the file. Once this is complete,
use 'pulumi stack import' to import the repaired stack.

refusing to proceed
```

## Expected Behavior

Editing the stack in this case may help, in other more complex cases it does not since dependencies will make it impracticable. since as a user I did not do anything wrong I would expect that pulumi knows how to handle this error and does not require me to perform manual steps that as mentioned not always are possible. If this was a transient error then pulumi should retry the deletion of the namespace at least until it is proven that the namespace for some unknown reason cannot be deleted. 

When reaching this case manual intervention would be required, but not by editing the stack, instead by manually deleting the namespace in kubernetes and then resuming pulumi destroy.

**Comment by manast** [comment]
Note, there may be another issue in pulumi that lead to the error ```[403] The provided update token has expired.``` which I have actually seen several times before, but the issue I am reporting is more general not only for this particular example.

**Comment by leezen** [comment]
Could you please provide verbose logs for this happening? https://www.pulumi.com/docs/troubleshooting/

The update token error you're seeing is a bit unexpected.

**Comment by manast** [comment]
This does not happen every time, and when it happens then it is too late to get the logs... only thing I know is that when this happens it usually has taken many minutes before failing. So whatever token that is expired maybe does because it took more time than expected to delete the resource. Anyway, if I can get more logs I will upload them.

**Comment by lukehoban** [comment]
Closing out as a dupe of #7094 which has a little bit of additional detail on what the expected problem is here.
